README.md

```diff
@@ -171,6 +171,7 @@ The options table has the following fields:
 * `pool`: custom connection pool name. Option as per [OpenResty docs](https://github.com/openresty/lua-nginx-module#tcpsockconnect), except that the default will become a pool name constructed using the SSL / proxy properties, which is important for safe connection reuse. When in doubt, leave it blank!
 * `pool_size`: option as per [OpenResty docs](https://github.com/openresty/lua-nginx-module#tcpsockconnect)
 * `backlog`: option as per [OpenResty docs](https://github.com/openresty/lua-nginx-module#tcpsockconnect)
+* `pool_only_after_response`: when enabled, set_keepalive() verifies http response fully read before pooling. Default is `false`
```

Comment:
is there a use case for this option? when would one ever want to pool a connection not fully read/in an undetermined state?

Reply:
@Tieske thanks for reviewing.
Actually this feature is meant to validate you don't have application-level bugs that lead to incorrect socket handling.

We have a large complex OpenResty-based system that pools connections in different code flows, and due to a bug we pooled connections in the wrong state (before reading the HTTP response). OpenResty's cosocket agreed to do so since the response bytes weren't received yet (race condition).
Using this flag puts a protocol-based validation that is enforced no matter how your application code behaves.

Reply:
Not sure I understand, but the reason you give is the reason for the overall PR right? you don't want to pool a socket that didn't complete the request cycle. I get that.

My comment; it should not be configurable using this flag `pool_only_after_response`, it should always close sockets from incomplete request cycles. No need for the option. My 2cts.

Reply:
@Tieske yeah it's the overall reason, to make sure set_keepalive() isn't called during a bad state.

I completely agree this should be always enforced, regardless of an option. I thought it would be easier to first introduce it under an optional option and after feeling confident enough - removing it. I've been using this option for a while and it seems to be working great in my environment. If you think I should remove it for this review - let me know and I will.

Reply:
I guess that's for @pintsized to judge. But as stated, imo it shouldn't be an option

Reply:
@pintsized let us know, I'll change accordingly. In any case - this protection is really helping me in my setup.

Reply:
for the record; I've run into this several times as well, so definitely a great fix.

Reply:
Hi both, thanks for looking at this. Sorry, I haven't really had time to look properly - yes I agree this doesn't seem to need a config option.